tests/fov_watcher_test.py

```diff
@@ -64,9 +72,22 @@ def _slow_copy_sample_tissue_data(
                 time.sleep(delta)
                 copied_tissue_path = os.path.join(dest, "." + tissue_file + ".aBcDeF")
                 os.rename(copied_tissue_path, os.path.join(dest, tissue_file))
+                num_bin_files += 1
             else:
                 shutil.copy(tissue_path, dest)
 
+    # get all .bin files
+    bin_files = [bfile for bfile in sorted(os.listdir(COMBINED_DATA_PATH)) if ".bin" in bfile]
+
+    # simulate updating the creation time for some .bin files, this tests _check_bin_updates
+    for i, bfile in enumerate(bin_files):
+        if i % 2 == 0:
+            shutil.copy(
+                os.path.join(COMBINED_DATA_PATH, bfile), os.path.join(dest, bfile + ".temp")
+            )
+            os.remove(os.path.join(dest, bfile))
+            os.rename(os.path.join(dest, bfile + ".temp"), os.path.join(dest, bfile))
```

Comment:
This should trigger `_check_bin_updates()`, but do you check anywhere that this is actually executed? Since the first extraction already generates an output tiff (albeit an incorrect one), I think you need an additional check for the new function beyond the existence of the output images.

For example you could test that the console does in fact print `f"Re-extracting incompletely extracted FOV {fov}"`.

Reply:
Good catch.